src/TextForEvent.js

```diff
@@ -9,7 +9,16 @@ function textForMemberEvent(ev) {
     ) : "";
     switch (ev.getContent().membership) {
         case 'invite':
-            return senderName + " invited " + targetName + ".";
+            var threePidContent = ev.getContent().third_party_invite;
+            if (threePidContent) {
+                // TODO: When we have third_party_invite.display_name we should
+                // do this as "$displayname received the invitation from $sender"
+                // or equiv
+                return targetName + " received an invitation from " + senderName;
+            }
+            else {
+                return senderName + " invited " + targetName + ".";
```

Comment:
The latter (normal invites) involves the sender making an action. The former (conversion of 3PID invite to normal invite) happens entirely passively and is divorced from any interaction from `senderName`. The wording is an attempt to reflect that passivity.